app/models/keyword.rb

```diff
@@ -4,6 +4,9 @@ class Keyword < ApplicationRecord
   enum status: { pending: 'pending', processing: 'processing', succeeded: 'succeeded', failed: 'failed' }
 
   has_many :keyword_search_entries, dependent: :destroy
+  has_one :keyword_stats, dependent: :nullify
+
+  delegate :top_ads_count, :total_ads_count, :non_ads_count, :top_ads_urls, :non_ads_urls, to: :keyword_search_entries_query
```

Comment:
It is a bit unexpected to delegate to a query object 🤔 It works, and to some extent, it makes sense, but usually, the only objects interacting with models are decorations 💡 

An alternative is not coupling the query in the model and limiting the usage to the controller 💡 

> [!NOTE]
> There is also https://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-extending to add new query methods

Reply:
Your suggestion does sound better. I remove the delegation logic in 5cb853aa0b9475d28e5e71fd1ccbb5927bb13403

The logic will be move to controller in the frontend PR